src/Network/Network/ApplicationGateway/NewAzureApplicationGatewayCommand.cs

```diff
@@ -91,6 +91,12 @@ public class NewAzureApplicationGatewayCommand : ApplicationGatewayBaseCmdlet
              HelpMessage = "The list of trusted root certificates")]
         public PSApplicationGatewayTrustedRootCertificate[] TrustedRootCertificate { get; set; }
 
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipelineByPropertyName = true,
+             HelpMessage = "The list of trusted client CA certificate chains")]
+        public PSApplicationGatewayTrustedClientCertificate[] TrustedClientCertificates { get; set; }
```

Comment:
please rename TrustedClientCertificates as TrustedClientCertificate,
powershell use singularity for parameter names:
https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/cmdlet-best-practices.md

Reply:
This is consistent with other parameter names in  NewAzureApplicationGatewayCommand, like SslCertificates, HttpListeners, FrontendPorts, etc., which take a list of PSObjects. Also, the naming TrustedClientCertificates makes it clear to the customer that this parameter takes a list of PSObjects verses a parameter like SslPolicy which takes on a single PSObject.
https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.commands.network.newazureapplicationgatewaycommand?view=azurerm-ps

Reply:
Yes, this parameter takes a list of objects, but it _also_ takes one signle object, aka `-TrustedClientCertificates @{...}` is totally OK. That is why we ask everyone to use singular name.

Considering exisiting parameters I won't request you to revise, but I do hope you understand why.